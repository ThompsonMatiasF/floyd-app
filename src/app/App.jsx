import React from 'react'
import '@axios-mock-db'
import StoreProvider from 'app/store'
import { Layout } from 'app/layout'
import moment from 'moment'
import 'moment/locale/es'

moment.locale('es')

const App = () => (
  <StoreProvider>
    <Layout />
  </StoreProvider>
);

export default App
