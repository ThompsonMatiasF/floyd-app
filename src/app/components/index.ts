export { default as ContainerCity } from '../containers/ContainerCity'
export { default as ContainerCurrentCity } from '../containers/ContainerCurrentCity'
export { default as City } from './City'
export { default as Weather } from './Weather'
export { default as WeatherIcon } from './WeatherIcon'
export { default as Forecast } from './Forecast'
export { default as ForecastDay } from './ForecastDay'
export { default as Tabs } from './Tabs'
export { default as Tab } from './Tab'
