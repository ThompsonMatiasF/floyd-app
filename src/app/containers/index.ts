export { default as ContainerCity } from './ContainerCity'
export { default as ContainerCurrentCity } from './ContainerCurrentCity'
