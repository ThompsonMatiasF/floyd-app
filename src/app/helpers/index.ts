export { default as constants } from './constants'
export { default as cityWeatherBuilder } from './weatherBuilder'
export { weatherClient } from './weatherClient'
