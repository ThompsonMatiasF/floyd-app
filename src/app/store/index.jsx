import React, { useReducer } from 'react'
import rootReducer from './reducers'

export const AppContext = React.createContext()

const StoreProvider = ({ children }) => {
  const [state, dispatch] = useReducer(...rootReducer);
  return <AppContext.Provider value={{state, dispatch}}>{children}</AppContext.Provider>
}

export default StoreProvider
