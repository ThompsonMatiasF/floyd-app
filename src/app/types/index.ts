export interface Temperature {
  main: number
  min: number
  max: number
}

export interface Weather {
  date: number
  description: string
  icon: string
  temp: Temperature
}

export interface Coordinates {
  lat: number
  lon: number
}

export interface Country {
  id: string
  name: string
}

export interface Location {
  id: number
  country: Country
  city: string
  coord: Coordinates
}

export interface CityWeather {
  location: Location
  weather: Weather
  forecast: Weather[]
}

export interface RequestState {
  LOADING: string
  SUCCESS: string
  FAILED: string
}

export interface WeatherActions {
  CITY: RequestState
  CURRENT_CITY: RequestState
  ACTIVE_TAB: string
}
