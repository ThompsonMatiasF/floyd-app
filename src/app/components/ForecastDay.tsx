import React, { FC } from 'react'
import moment from 'moment'
import { glassCard } from 'app/helpers/constants'
import { Weather } from 'app/types'
import { WeatherIcon } from './'

interface ForecastDayProps {
  weather: Weather
}

const ForecastDay: FC<ForecastDayProps> = ({ weather }) => {
  return (
    <div
      className="w-full md:w-auto flex flex-row items-center my-4"
      style={{
        borderRadius: '50px',
        ...glassCard,
      }}
      data-testid="forecast-day"
    >
      <div
        className="bg-blue-500"
        style={{
          borderRadius: '50px',
        }}
      >
        <WeatherIcon icon={weather.icon} description={weather.description} />
      </div>
      <div className="text-center flex-1 pl-4 pr-6 text-xl md:text-base">
        <div className="inline-block md:block capitalize">{moment.unix(weather.date).format('dddd DD')}</div>
        <span className="inline md:hidden"> - </span>
        <div className="inline-block md:block"><span className="font-bold">{weather.temp.min}ยบ</span> / <span className="font-bold">{weather.temp.max}ยบ</span></div>
      </div>
    </div>
  )
}

export default ForecastDay
