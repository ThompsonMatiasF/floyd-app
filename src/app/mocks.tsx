import React, {FC} from 'react'
import { CityWeather, Location, Weather } from 'app/types'
import { initialState } from './store/reducers/weather/weather.reducer'
import { AppContext } from 'app/store';

export const weatherMock: Weather = {
  date: 1610553600,
  description: 'cielo claro',
  icon: '01d',
  temp:  {
    main: 26,
    min: 25,
    max: 27,
  },
}

export const locationMock: Location = {
  id: 3435010,
  country: {
    id: 'AR',
    name: 'Argentina',
  },
  city: 'Don Torcuato',
  coord: {
    lat: -34.4924,
    lon: -58.6162,
  },
}

export const cityWeatherMock: CityWeather = {
  location: locationMock,
  weather: weatherMock,
  forecast: Array(5).fill(weatherMock),
}

const state = {
  weather: {
    ...initialState,
    city: {
      ...initialState.city,
      data: cityWeatherMock,
    },
    currentCity: {
      ...initialState.currentCity,
      data: cityWeatherMock,
    },
  },
}

export const MockStoreProvider: FC = ({ children }) => (
  <AppContext.Provider value={{state, dispatch: jest.fn()}}>{ children }</AppContext.Provider>
)
