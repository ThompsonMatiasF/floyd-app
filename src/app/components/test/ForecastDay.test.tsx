import { render } from '@testing-library/react'
import ForecastDay from '../ForecastDay'
import { weatherMock } from 'app/mocks'

describe('ForecastDay.tsx', () => {
  it('should match snapshot', () => {
    const container = render(<ForecastDay weather={weatherMock} />).container

    expect(container).toMatchSnapshot()
  })
})
