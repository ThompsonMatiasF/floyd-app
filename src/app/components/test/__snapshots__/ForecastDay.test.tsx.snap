// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDay.tsx should match snapshot 1`] = `
<div>
  <div
    class="w-full md:w-auto flex flex-row items-center my-4"
    data-testid="forecast-day"
    style="border-radius: 50px;"
  >
    <div
      class="bg-blue-500"
      style="border-radius: 50px;"
    >
      <img
        alt="cielo claro"
        src="http://openweathermap.org/img/wn/01d.png"
      />
    </div>
    <div
      class="text-center flex-1 pl-4 pr-6 text-xl md:text-base"
    >
      <div
        class="inline-block md:block capitalize"
      >
        Wednesday 13
      </div>
      <span
        class="inline md:hidden"
      >
         - 
      </span>
      <div
        class="inline-block md:block"
      >
        <span
          class="font-bold"
        >
          25
          ยบ
        </span>
         / 
        <span
          class="font-bold"
        >
          27
          ยบ
        </span>
      </div>
    </div>
  </div>
</div>
`;
